lib/io.toit

```diff
@@ -732,7 +784,9 @@ abstract class Reader implements old-reader.Reader:
   ```
   */
   little-endian -> EndianReader:
-    return EndianReader --reader=this --byte-order=binary.LITTLE_ENDIAN
+    if not endian_ or endian_.byte-order_ != binary.LITTLE_ENDIAN:
```

Comment:
I wouldn't read the field that many times.

Reply:
done.